### PR TITLE
Add new events onClientVehicleCreate and onVehicleCreate

### PR DESCRIPTION
**Pull request by syncc01**
These events are called when a vehicle is created.

I have seen that a group of people have requested this event in our wonderful MTA and I have decided to add them. 

This is my first pull request, good luck to me (:

### CODE REVIEW

**Comment by CrosRoad95** [comment]
https://github.com/multitheftauto/mtasa-blue/pull/218 - read comments,
make "vehicle factory" on your server that going to trigger certain event

File: Client/mods/deathmatch/logic/CClientGame.cpp [hunk 1]
```diff
@@ -2528,6 +2528,7 @@ void CClientGame::AddBuiltInEvents()
     m_Events.AddEvent("onClientPedStep", "foot", nullptr, false);
 
     // Vehicle events
+    m_Events.AddEvent("onClientVehicleCreate", "", NULL, false);
```
Comment (Lpsd) [inline Client/mods/deathmatch/logic/CClientGame.cpp]: The event should be called `onClientVehicleCreated`

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 2]
```diff
@@ -2806,11 +2806,14 @@ CClientVehicle* CStaticFunctionDefinitions::CreateVehicle(CResource& Resource, u
         if (ucVariant2 == 255 && ucVariant == 255)
             CClientVehicleManager::GetRandomVariation(usModel, ucVariation, ucVariation2);
 
-        CClientVehicle* pVehicle = new CDeathmatchVehicle(m_pManager, NULL, INVALID_ELEMENT_ID, usModel, ucVariation, ucVariation2);
+        CClientVehicle* pVehicle = new CDeathmatchVehicle(m_pManager, NULL, INVALID_ELEMENT_ID, usModel, ucVariation, ucVariation2);;
```
Comment (Lpsd) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: Why is there an extra semicolon here?

**Review by Lpsd — COMMENTED** [review]
Left some comments. I think we should consider some of the points raised in #218 - although this event has less scope (vehicles only), the event system is still quite slow. 

Some performance tests would be nice.

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp (hunk 2 above)
Comment (syncc01) [reply Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: oh, it was a mistake

**Comment by CrosRoad95** [comment]
it should be rejected because of the same reason as in #218
add "onElementCreated" or nothing, perhaps mta should have some kind of separete event system for frequently called events

File: Client/mods/deathmatch/logic/CClientGame.cpp (hunk 1 above)
Comment (syncc01) [reply Client/mods/deathmatch/logic/CClientGame.cpp]: Done

**Pull request closed without merging**